app/src/main/java/com/example/drawingApp/ColorPickerUtility.kt

```diff
@@ -0,0 +1,222 @@
+package com.example.drawingApp
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Color
+import android.view.LayoutInflater
+import android.widget.TextView
+import androidx.core.graphics.toColor
+import androidx.core.widget.addTextChangedListener
+import com.google.android.material.bottomsheet.BottomSheetDialog
+import kotlin.math.max
+import kotlin.math.min
+
+object ColorPickerUtility {
+
+    @SuppressLint("ClickableViewAccessibility")
+    fun colorPickerSheet(
+        context: Context,
+        /**
+         * Called whenever the hue is updated via text or slider.
+         * Use it to set wherever you are storing your hue
+         * and then update your color
+         */
+        onHueUpdate: (Float) -> Unit,
```

Comment:
Why do you need both these lambdas? Why not just a "onColorUpdate"? Does anyone outside the scope of this sheet need individual updates?

Reply:
removed and incorported an onColorUpdate